### ISSUE TITLE
Suggestion: new container with AAD use authenticatioemail as user name

### ISSUE THREAD

**Issue by ti-jalopez** [microsoft/navcontainerhelper#2300]
PLEASE DO NOT INCLUDE ANY PASSWORDS OR TOKENS IN YOUR ISSUE!!!

**Describe the issue**
When I create a new container using New-BcContainer using auth AAD and authenticationemail, I pass parameters $credential for container user and e-mail for ADD authentication

The user created in BC in CRONUS database is $credential.UserName with $authenticationemail.
And the user needed to access container (windows user) is $credential.UserName using $credential.Password.
In this situation, the User Name in BC is the same as the container, but other than the authentication email.

I suggest the in BC User Name should be the same as Authentication Email.
In this case, we have separate credentials for:
* Authenticate in container or publish apps from VSCode as NavUser.
* Access to BC using AAD credentials using as User Name the same as Authentication Email.

**Scripts used to create container and cause the issue**
Example: 
```
 $credential.UserName = ADMIN
 $authenticationemail = myaccount@domail.com
```

**Full output of scripts**
In BC the user created is:
```
User Name: ADMIN
Authentication Email: myaccount@domail.com
```
For container access or VSCode access by "authentication": "UserPassword"
```
User Name: ADMIN
Password:  (the password used to create container)
```

**Suggested change**
In BC the user created is:
```
User Name: myaccount@domail.com
Authentication Email: myaccount@domail.com
```
For container access or VSCode access by "authentication": "UserPassword"
```
User Name: ADMIN
Password:  (the password used to create container)
```

**Screenshots**

**Additional context**
- does it happen all the time?
- did it use to work?

**Comment by freddydk** [comment]
So you want to use username/password authentication with the AAD username as username and the same password as password?
Why wouldn't you just use AAD authentication from VS Code (which works with the latest changes in BcContainerHelper)

I am not a big believer of using your AAD password other places than in a secure AAD login environment (but I might have misunderstood your suggestion)

**Comment by ti-jalopez** [comment]
I'm using many containers shared in a traefik server for many developers and testers using default Cronus database.
Until now, I was always using UserPassword to access from VSCode.
My first idea for containers using AAD is to create a container than can be used with AAD authentication for users in web client, and the two types of authentication from VSCode (UserPassword -that use common for anypeole- or AAD -new in VSCode from OnPrem containers-).
With my initial tests (see description of this issue) I've tried to create a container with two users: one for AAD and other for NavUserPassword, but I have had problems using the admin user (the one I want for NavUserPassword from VSCode) and the email, since they are different users.
Finally, after thinking about Freddy's answer and researching the New-NavContainerHelper code, I found a solution to create both users, I've modified my script for:

1.  Create a container using the same UserName for container admin as the authentication email for AAD. It creates a containers with one user where the user name is the same as authentication email. 
2. Create a user of type NavUserPassord that is common and shared by all developers.

After that, developers can connect from VSCode using dual authentication types.

Finally, I think my suggestion isn't a good idea to modify bccontainerhelper. 
@freddydk , yo can close this issue. Thanks for all.
